genai-perf/genai_perf/parser.py

```diff
@@ -135,47 +162,38 @@ def _check_conditional_args(
     Check for conditional args and raise an error if they are not set.
     """
 
-    # Endpoint and output format checks
-    if args.service_kind == "openai":
-        if args.endpoint_type is None:
-            parser.error(
-                "The --endpoint-type option is required when using the 'openai' service-kind."
-            )
-        else:
-            if args.endpoint_type == "chat":
-                args.output_format = ic.OutputFormat.OPENAI_CHAT_COMPLETIONS
-            elif args.endpoint_type == "completions":
-                args.output_format = ic.OutputFormat.OPENAI_COMPLETIONS
-            elif args.endpoint_type == "embeddings":
-                args.output_format = ic.OutputFormat.OPENAI_EMBEDDINGS
-            elif args.endpoint_type == "rankings":
-                args.output_format = ic.OutputFormat.RANKINGS
-            elif args.endpoint_type == "image_retrieval":
-                args.output_format = ic.OutputFormat.IMAGE_RETRIEVAL
-
-            # (TMA-1986) deduce vision format from chat completions + image CLI
-            # because there's no openai vision endpoint.
-            elif args.endpoint_type == "vision":
-                args.output_format = ic.OutputFormat.OPENAI_VISION
-            elif args.endpoint_type == "nvclip":
-                args.output_format = ic.OutputFormat.NVCLIP
-
-            if args.endpoint is not None:
-                args.endpoint = args.endpoint.lstrip(" /")
-            else:
-                args.endpoint = _endpoint_type_map[args.endpoint_type]
-    elif args.endpoint_type is not None:
+    if args.endpoint_type not in _endpoint_type_map:
+        parser.error(f"Invalid endpoint type {args.endpoint_type}")
+        return
```

Comment:
Do we need to return? It looks like [ArgumentParser.error](https://docs.python.org/3/library/argparse.html#argparse.ArgumentParser.error) terminates the program.

Reply:
that is correct - but I followed the pattern in the file - 

Would want to keep it consistent - but generally if you use parser.error you don't need a return - but maybe its clearer this way?